### PR TITLE
Send build information in property bag for Helix

### PR DESCRIPTION
**Pull request by rokonec**

### CODE REVIEW

File: src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs [hunk 1]
```diff
@@ -202,6 +201,12 @@ protected override async Task ExecuteCore(CancellationToken cancellationToken)
                     }
                 }
 
+                def = AddBuildVariableProperty(def, "BuildNumber", "Build.BuildNumber");
```
Comment (safern) [inline src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:204]: Wouldn't this have been simpler to use `HelixProperties` item group in MSBuild? We do that for other properties:

https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/tools/Microsoft.DotNet.Helix.Sdk.MonoQueue.targets#L35-L37

Comment (safern) [reply src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:204]: Plus it is weird to have the property bag item values defined over multiple places, makes it harder to find what properties are passed, that is why I didn't find this was already done in arcade until @MattGal pointed me to it, I was about to send a PR for this.

Comment (rokonec) [reply src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:204]: Although people could have added it themselves they mostly do not. 
There were requests from partners to have such data by default, as we believe that additional diagnostics data rarely harm we have fulfilled their requests.

Comment (safern) [reply src/Microsoft.DotNet.Helix/Sdk/SendHelixJob.cs:204]: I meant adding them via MSBuild property ItemGroup in the helix SDK targets, which all repos use. Just as the configuration and os properties I linked above, which are in the SDK targets and all jobs get it by default.

**Pull request merged**